### PR TITLE
TRE-655 migration guide

### PR DESCRIPTION
**Pull request by ameliemaucade**

### CODE REVIEW

File: v2-to-v3-migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+### New price representation 
+
+The Recurring API V3 introduces a new JSON representation for agreement price.
+
+There is no change on the charge limits rules. Going forward, we will look into how we can implement charge limits in a better way, that takes care of the merchants needs.
+
+To draft an agreement with a fix amount with the same charge limit as in V2, `pricing.type` should be set to `LEGACY`.
+
+**Note**: `pricing.type` is an optional field. If not provided in the request the type will be defaulted to `LEGACY`.
```
Comment (rognlien) [inline v2-to-v3-migration-guide.md:49]: fix -> fixed?

**Review by rognlien — APPROVED** [review]

File: v2-to-v3-migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+The API V3 returns different response status for some endpoints:
+
+| Endpoint                                                                               | V2 response status | V3 response status                  |
+|----------------------------------------------------------------------------------------|--------------------|-------------------------------------|
+| [`PATCH:/agreements/{agreementId}`][update-agreement-endpoint]                         | `200 OK`           | `204 No Content` or `202 Accepted`* | 
+| [`DELETE:/agreements/{agreementId}/charges/{chargeId}`][cancel-charge-endpoint]        | `200 OK`           | `204 No Content` or `202 Accepted`* | 
+| [`POST:/agreements/{agreementId}/charges/{chargeId}/capture`][capture-charge-endpoint] | `200 OK`           | `204 No Content` or `202 Accepted`* | 
+| [`POST:/agreements/{agreementId}/charges/{chargeId}/refund`][refund-charge-endpoint]   | `200 OK`           | `204 No Content` or `202 Accepted`* | 
+
+**Note** In the future the API might return `202 Accepted` for some operations if the request is processed asynchronously.
```
Comment (ameliemaucade) [inline v2-to-v3-migration-guide.md:33]: add text on retry-after header

File: v2-to-v3-migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    "amount": 100000,
+    "currency": "NOK"
+  },
+  "productName": "MyNews Digital",
+  "customerPhoneNumber": "45678272",
+  "...": "..."
+}
+```
+
+To draft agreement with a [variable amount](https://vippsas.github.io/vipps-developer-docs/docs/APIs/recurring-api/vipps-recurring-api#recurring-agreements-with-variable-amount), `pricing.type` should be set to `VARIABLE`.
```
Comment (ameliemaucade) [inline v2-to-v3-migration-guide.md:94]: add section for variable amount

File: v2-to-v3-migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+## New price representation 
+
+The Recurring API V3 introduces a new JSON representation for agreement price.
+
+There is no change on the charge limits rules. Going forward, we will look into how we can implement charge limits in a better way, that takes care of the merchants needs.
+
+To draft an agreement with a fixed amount with the same charge limit as in V2, `pricing.type` should be set to `LEGACY`.
+
+**Note**: `pricing.type` is an optional field. If not provided in the request the type will be defaulted to `LEGACY`.
```
Comment (ameliemaucade) [inline v2-to-v3-migration-guide.md:65]: fix typo

File: v2-to-v3-migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+      "success": true
+    }
+  ]
+}
+````
+
+## Idempotency key
+The misspelled `Idempotent-Key` header is deprecated.
+The`Idempotency-Key` header is now required for the `POST` and `PATCH` endpoints. 
+See [Idempotency key header](https://vippsas.github.io/vipps-developer-docs/docs/APIs/recurring-api/idempotency-key-header-v3-api-coming-soon) in the API Guide.
```
Comment (ameliemaucade) [inline v2-to-v3-migration-guide.md:211]: fix link

**Review by rebekaburnett — APPROVED** [review]
LGTM! :-)

**Pull request merged**